internal/pkg/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+// Package dockerfile provides simple Dockerfile parsing functionality
+package dockerfile
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+var exposeRegexPattern = regexp.MustCompile(`(?P<port>\d+)(\/(?P<protocol>\w+))?`) // port and optional protocol, at least 1 time on a line
+
+const (
+	exposeRegexpWholeMatch = 0
+	exposeRegexpPort       = 1
+	exposeRegexpProtocol   = 3
+)
+const reFindAllMatches = -1 // regexp package uses this as shorthand for "find all matches in string"
+
+var (
+	errCouldntParseDockerfilePort = errors.New("parse port from EXPOSE")
+)
+
+type portConfig struct {
+	Port      uint16
+	Protocol  string
+	RawString string
+	err       error
+}
+
+// Dockerfile represents a parsed dockerfile
+type Dockerfile struct {
+	ExposedPorts []portConfig
+	parsed       bool
+	path         string
+
+	fs afero.Fs
+}
+
+// New returns an empty Dockerfile
+func New(fs afero.Fs, path string) *Dockerfile {
```

Comment:
I think it'd be cool if we took in an IO.Reader here, instead of the filesystem and string and have app do the locating and reading of this file. That'd be cool because in the future we could maybe inspect a dockerfile from someone's remote repo and analyze it :D

Reply:
I think I'd like to [punt on this](https://github.com/aws/amazon-ecs-cli-v2/issues/832) for the future. There's now an issue in the backlog for a future pr. Does that make sense for the time being?